meilisearch-schema/src/position_map.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::BTreeMap;
+
+use crate::{FieldId, IndexedPos};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct PositionMap {
```

Comment:
Could you describe a little more what's the purpose of this struct, what is it used for?

Reply:
What I understand is that you exported and fixed the logic from the Schema into this struct. It is a struct that maps between displayed fields ids and searchable fields ids.

Reply:
This DS maps a field ID to an indexed position. It contains ALL the fields that have been found in documents, and serves as the source of truth for the `*` in both searchable and displayed attributes.